src/document/generator.ml

```diff
@@ -1430,14 +1430,22 @@ module Make (Syntax : SYNTAX) = struct
           ++ Link.from_path (m :> Paths.Path.t)
           ++ O.txt " " ++ O.keyword "end"
 
+    and is_elidable_with_u : Odoc_model.Lang.ModuleType.U.expr -> bool = function
```

Comment:
Why not render nested substitutions ? It is rendered for `T`:

```
# Module Test
module type S = sig ... end
module type T = S with type M.t := int
```

but the documentation for `T` doesn't mention any substitution at all:
```
# Module type Test.T
module M : sig ... end
type t = int
```

Showing the substitution is not that bad:

```
# Module type Test.T
module M : M.S with type t := int
type t = int
```

It would be even more useful in case of non-destructive substitution because the definition of `t` would be in the same documentation page:
```
# Module type Test.T
module M : M.S with type t = int
type t = M.t
```

Reply:
Hmm, it was certainly not my intention to remove the 'with type' always, just the construct `sig ... end with type ...`

I'll double check.

Reply:
I think I misinterpreted what you mean - though I'm still not quite sure! To be clear, we've got:

```ocaml
module type S = sig
  module M: sig
    type t
  end
  type t = M.t
end
module type T = S with type M.t := int
```

You wrote that you thought this would be better:

```ocaml
# Module type Test.T
module M : M.S with type t := int
type t = int
```

I don't think you mean `M.S` - I think you might have meant `S.M` - but we can't do this because we can't have paths to elements within module types. Is this correct?

Reply:
By `M.S`, I meant `S.M` or even `Test.S.M`, sorry for the confusion.

> Hmm, it was certainly not my intention to remove the 'with type' always, just the construct `sig ... end with type ...`

`sig ... end with type ...` is turned into just `sig ... end` but I'd expect `Test.S.M with type ...` even if `M` is not a module type.

> but we can't do this because we can't have paths to elements within module types.

On the definition of `T`, we have the path to `S` and the substitution, why can't nested substitution be represented the same way ?
Perhaps we can represent it like `(module type of Test.S.M) with type ...` internally ?

Reply:
The problem isn't `M`, it's `S` - `S` is a module type so we can't do `S.M`.

Reply:
That wouldn't be valid OCaml but it would be a valid reference (so can work internally). Do we require that every rendered paths should be valid OCaml paths ?

Reply:
Even internally it's a Path not a Reference, and in any case I strongly prefer sticking to valid OCaml than to deviate. If the language supports it some day we can do it - for example I'm keeping an eye on the transparent ascription PR as that will help with some tricker parts of handling `module type of`.

Reply:
Alright :) Sorry for the digression.